plugins/web/opentelemetry-instrumentation-web-exception/src/instrumentation.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+import {
+  ATTR_EXCEPTION_TYPE,
+  ATTR_EXCEPTION_MESSAGE,
+  ATTR_EXCEPTION_STACKTRACE,
+} from '@opentelemetry/semantic-conventions';
+// TODO: should events be imported or included as part of the instrumentation base class?
+import { events } from '@opentelemetry/api-events';
+import { SeverityNumber } from '@opentelemetry/api-logs';
+import { Attributes } from '@opentelemetry/api';
+
+export interface GlobalErrorsInstrumentationConfig
+  extends InstrumentationConfig {
+  /**
+   * A callback function for adding custom attributes to the span when an error is recorded.
+   *
+   * @param {Error} error - The error object that is being recorded.
+   * @returns {Attributes} - The attributes to add to the span.
+   */
+  applyCustomAttributes?: (error: Error) => Attributes;
+}
+
+export class WebExceptionInstrumentation extends InstrumentationBase<GlobalErrorsInstrumentationConfig> {
+  readonly applyCustomAttributes?: (error: Error) => Attributes;
+  constructor(config: GlobalErrorsInstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-web-exception', '0.0.1', config);
+
+    this.applyCustomAttributes = config.applyCustomAttributes;
+  }
+
+  init() {}
+
+  onError = (event: ErrorEvent | PromiseRejectionEvent) => {
+    const error: Error | undefined =
+      'reason' in event ? event.reason : event.error;
+    if (error) {
+      const message = error.message;
+      const type = error.name;
+      const errorAttributes = {
+        [ATTR_EXCEPTION_TYPE]: type,
+        [ATTR_EXCEPTION_MESSAGE]: message,
+        [ATTR_EXCEPTION_STACKTRACE]: error.stack,
+      };
+
+      const eventLogger = events.getEventLogger(
```

Comment:
We should also be including the "parsed" filename as well as the current url (hosting page), I also left a [comment in](https://github.com/open-telemetry/semantic-conventions/pull/1942/files#r1989711556) that this should be part of the client exception

Reply:
Note: if we end up using the `url.full` then to [comply with this PR](https://github.com/open-telemetry/semantic-conventions/pull/971) it MUST be redacted and I don't believe that the JS repo currently has any "helpers" to construct a redacted `url.full`.

Reply:
Additional comment, while it would be "nice" to just say that the URL is in the "resources" for SPA's it really needs to be mutable which resources can't currently be 😢

Reply:
For now, since we're sticking with the existing semantic conventions instead of client specific ones I haven't included the parsed file name or URL. I think we can add the URL in a future PR and add some core helpers to do that separately.